pkg/pool-manager/macpoolmap.go

```diff
@@ -98,7 +98,7 @@ func (m *macMap) alignMacEntryAccordingToVmInterface(macAddress, instanceFullNam
 	logger := log.WithName("alignMacEntryAccordingToVmInterface")
 	macEntry, _ := m.findByMacAddress(macAddress)
 	for _, iface := range vmInterfaces {
-		if iface.MacAddress == macAddress {
+		if NewMacKey(iface.MacAddress).normalizedMacAddress == macAddress {
```

Comment:
can we use the `String` method?
```suggestion
		if NewMacKey(iface.MacAddress).String() == macAddress {
```

I think it works, see [here](https://go.dev/play/p/c0PoTSrbmF1)